examples/pytorch_lightning_simple.py

```diff
@@ -0,0 +1,179 @@
+        input_dim = 28 * 28
+        for i in range(n_layers):
+            output_dim = int(trial.suggest_loguniform('n_units_l{}'.format(i), 4, 128))
+            self.layers.append(nn.Linear(input_dim, output_dim))
+            self.dropouts.append(nn.Dropout(dropout))
+            input_dim = output_dim
+
+        self.layers.append(nn.Linear(input_dim, CLASSES))
+
+        # Assigning the layers as class variables (PyTorch requirement).
+        for idx, layer in enumerate(self.layers):
+            setattr(self, 'fc{}'.format(idx), layer)
+
+        # Assigning the dropouts as class variables (PyTorch requirement).
+        for idx, dropout in enumerate(self.dropouts):
+            setattr(self, 'drop{}'.format(idx), dropout)
+
+    # Forward pass computation function.
+    def forward(self, data):
+        data = data.view(-1, 28 * 28)
+        for layer, dropout in zip(self.layers, self.dropouts):
+            data = F.relu(layer(data))
+            data = dropout(data)
+        return F.log_softmax(self.layers[-1](data), dim=1)
+
+
+class LightningNet(pl.LightningModule):
+
+    def __init__(self, trial):
+        super(LightningNet, self).__init__()
+
+        # Be careful not to overwrite `pl.LightningModule` attributes such as `self.model`.
+        self._model = Net(trial)
+
+    def forward(self, data):
+        return self._model(data)
+
+    def training_step(self, batch, batch_nb):
+        data, target = batch
+        output = self.forward(data)
+        loss = F.nll_loss(output, target)
+        return {'loss': loss}
+
+    def validation_step(self, batch, batch_nb):
+        data, target = batch
+        output = self.forward(data)
+        pred = output.argmax(dim=1, keepdim=True)
+        correct = pred.eq(target.view_as(pred)).sum().item()
+        accuracy = correct / data.size(0)
+        return {'validation_accuracy': accuracy}
+
+    def validation_end(self, outputs):
+        accuracy = sum(x['validation_accuracy'] for x in outputs) / len(outputs)
+        # Pass the accuracy to the `DictLogger` via the `'log'` key.
+        return {'log': {'accuracy': accuracy}}
+
+    def configure_optimizers(self):
+        return Adam(self._model.parameters())
+
+    @pl.data_loader
+    def train_dataloader(self):
+        return torch.utils.data.DataLoader(
+            datasets.MNIST(DIR, train=True, download=True, transform=transforms.ToTensor()),
+            batch_size=BATCHSIZE, shuffle=True)
+
+    @pl.data_loader
+    def val_dataloader(self):
+        return torch.utils.data.DataLoader(
+            datasets.MNIST(DIR, train=False, download=True, transform=transforms.ToTensor()),
+            batch_size=BATCHSIZE, shuffle=False)
+
+
+def objective(trial):
+    # PyTorch Lightning will try to restore model parameters from previous trials if checkpoint
+    # filenames match. Therefore, the filenames for each trial must be made unique.
+    checkpoint_callback = pl.callbacks.ModelCheckpoint(
```

Comment:
The filenames may conflict if this example script is executed multiple times as follows:
```console
$ python pytorch_lightning_simple.py
$ python pytorch_lightning_simple.py
```

How about removing the model files after the execution of `study.optimize(..)` is finished?
See also: [tensorflow_estimator_simple.py#L133](https://github.com/pfnet/optuna/blob/master/examples/tensorflow_estimator_simple.py#L133)

Reply:
Changes as you suggested. You can now rerun the same scripts multiple times.